pkgs/applications/misc/flashprint/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv
+, lib
+, fetchurl
+, dpkg
+, glibc
+, gcc-unwrapped
+, autoPatchelfHook
+, libGLU
+, qtbase
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation {
+  name = "flashprint";
+  version = "5.6.1";
+  src = fetchurl {
+    url = "https://en.fss.flashforge.com/10000/software/01e7742d5a721201cd33d4f8c083b63b.deb";
+    sha256 = "sha256-lvK4igv/L1zrqNtqbWtCyZrSZbkY8Ak/Li3NeVAH1YU=";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook dpkg wrapQtAppsHook ];
+  buildInputs = [ glibc gcc-unwrapped libGLU qtbase ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    dpkg -x $src $out
+    mv $out/usr/* $out/
+    rm -r $out/usr
+    ln -s $out/share/FlashPrint5/FlashPrint $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "The 3d printing slicer by Flashforge";
+    homepage = "https://www.flashforge.com/download-center/63";
+    license = licenses.unfree;
```

Comment:
You can specify `mainProgram = "FlashPrint";` so that when running with `nix run .#flashprint` it will know which binary to run.